src/SingleReport/BootMouse.cpp

```diff
@@ -73,18 +78,24 @@ int BootMouse_::getInterface(uint8_t* interfaceCount)
 
 int BootMouse_::getDescriptor(USBSetup& setup)
 {
-	// Check if this is a HID Class Descriptor request
-	if (setup.bmRequestType != REQUEST_DEVICETOHOST_STANDARD_INTERFACE) { return 0; }
-	if (setup.wValueH != HID_REPORT_DESCRIPTOR_TYPE) { return 0; }
-
 	// In a HID Class Descriptor wIndex cointains the interface number
 	if (setup.wIndex != pluggedInterface) { return 0; }
 
-	// Reset the protocol on reenumeration. Normally the host should not assume the state of the protocol
-	// due to the USB specs, but Windows and Linux just assumes its in report mode.
-	protocol = HID_REPORT_PROTOCOL;
+	// Check if this is a HID Class Descriptor request
+	if (setup.bmRequestType != REQUEST_DEVICETOHOST_STANDARD_INTERFACE) { return 0; }
```

Comment:
Why did you move this down?

Reply:
To make if-else branch for `setup.wValueH`. Previously, there was an inequality and a return, and now there is a check for two types of descriptor.

Reply:
Yes I saw this, but this check can be kept above as well, right? I mean it should not matter and I also do not prefer anything. It is just for my understanding, that this line could also be placed above the wIndex check, right?

Reply:
Yes, there is no difference. I put the index above for reasons that first you need to check whether there was a request to this device, and then proceed to check the type of request. It would be strange to check the type first, then the index, and then the other type.